### ISSUE TITLE
RangeControl with initial value cannot be changed using text field

### ISSUE THREAD

**Issue by pavel-mailpoet** [WordPress/gutenberg#24264]
**Describe the bug**
When I try to change a value in the RangeControl I cannot change the value using the text field. The value keeps resetting to the initial value.

**To reproduce**
1. Go to https://wordpress.github.io/gutenberg/?path=/story/components-rangecontrol--with-help
2. Click inside the text input field where is the initial value `5`
3. Press Backspace to clear the input field
4. Press `7` on your keyboard

**Expected behavior**
I expect to see the number `7` in the input field, but there is actually a number `5`.

**Screenshots**
You can also see the behaviour here: https://d.pr/i/yj0Zun I demonstrated the problem in the Gutenberg plugin itself. No matter what I try to enter into the input field it always uses the initial value.

**Editor version (please complete the following information):**
- WordPress version: 5.4.2
- It happens in the Gutenberg plugin and also in the Storybook.
- If the Gutenberg plugin is installed, which version is it? 8.6.0

**Desktop (please complete the following information):**
 - OS: MacOs
 - Browser: Firefox, Safari

**Comment by costasovo** [comment]
I found out that this issue was fixed by commit https://github.com/WordPress/gutenberg/pull/24251/commits/4cf23dcac353f0e46acf2339d98895e60c9e1662. Probably caused by the same thing as issue https://github.com/WordPress/gutenberg/pull/24251. I guess we just have to wait for the release of packages.

**Comment by cngodles** [comment]
Not sure if related, but the initial value is only populating into the control and not the number input field. I set a default value of 0.3 here, but the component that gets added doesn't pick it up unless I change it.

Screenshot of the code on load. Note the `.components-range-control__number` input doesn't have a value set.

![image](https://user-images.githubusercontent.com/5685764/89544029-a2567a80-d7cf-11ea-9ff6-df07b267e74a.png)

This is what I used in the JS for creating the control:

```
overlayOpactiy: {
      type: 'number',
      default: 0.3
    }
```
